cmd/lifecycle/analyzer.go

```diff
@@ -120,6 +120,30 @@ func (a *analyzeCmd) Privileges() error {
 		if err != nil {
 			return cmd.FailErr(err, "initialize docker client")
 		}
+	} else {
+		if a.platformAPIVersionGreaterThan06() {
+			err := a.verifyReadAccess(a.previousImage)
+			if err != nil {
+				return cmd.FailErr(err)
+			}
+			err = a.verifyReadAccess(a.runImageRef)
+			if err != nil {
+				return cmd.FailErr(err)
+			}
+			if len(a.additionalTags) > 0 {
+				for _, tag := range a.additionalTags {
+					err := a.verifyReadWriteAccess(tag)
+					if err != nil {
+						return cmd.FailErr(err)
+					}
+				}
+			}
+		} else {
+			err := a.verifyReadWriteAccess(a.platform06.cacheImageTag)
+			if err != nil {
+				return cmd.FailErr(err)
+			}
+		}
```

Comment:
We could reduce the complexity of this code by just iterating over `a.registryImages()`. It might also be worthwhile to make an `a.validateRegistryAccess` function, so that it could also be invoked by the creator.

Reply:
cc @jabrown85 for moving logic out of `cmd` package!

Reply:
Hmm. Yeah. The different scopes required make this kind of annoying to deal with. Maybe introducing a struct (or interface) that the `image` package could rely on. That way creator could pass `createCmd` and analyzer could pass `analyzeArgs `.


```
"github.com/buildpacks/lifecycle/image"

// the proposed struct, but we could also introduce an interface with methods like ReadableImages() WriteableImages() that were implemented in both creator and analyzer cmds/args
type registryInputs struct {
  cacheImagetag string
  previousImage string
  additionalTags []string
}

// in analyzer
if err := image.VerifyRegistryAccess(a); err != nil {
		return cmd.FailErr(err)
}

// in creator
if err := image.VerifyRegistryAccess(c); err != nil {
		return cmd.FailErr(err)
}


VerifyRegistryAccess(regInputs registryInputs) {...}
```

Reply:
Ohhh I forgot that `registryImages` could include images that you just need read access to, and ones that you need read/write access to... introducing a new struct might be a helpful way to organize this.

Reply:
@jabrown85 , @natalieparellano  I found useful the idea with the `interface` because it allows me to simplify the validation logic. It will look like this:

Reply:
I refactored the code in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/1ab080726bdfd9210f0b21fe75dae32a42697ef2)